pkgs/sops-install-secrets/main.go

```diff
@@ -509,6 +511,80 @@ func importSSHKeys(keyPaths []string, gpgHome string) error {
 	return nil
 }
 
+func handleRestart(symlinkPath string, secretDir string, secrets []secret) error {
+	var restart []string
+	var reload []string
+	hasher := sha256.New()
+	for _, secret := range secrets {
+		// Skip when there are no restarts/reload
+		if len(secret.RestartUnits) == 0 && len(secret.ReloadUnits) == 0 {
+			continue
+		}
+		oldPath := filepath.Join(symlinkPath, secret.Name)
+		newPath := filepath.Join(secretDir, secret.Name)
+
+		// Calculate the old hash
+		oldFile, err := os.Open(oldPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				// File did not exist before
+				restart = append(restart, secret.RestartUnits...)
+				reload = append(reload, secret.ReloadUnits...)
+				continue
+			}
+			return err
+		}
+		defer oldFile.Close()
+		if _, err = io.Copy(hasher, oldFile); err != nil {
```

Comment:
Is comparing both files byte-wise not easier and faster than calculating sha256 hashes? It's also 100% collision free :)